### PR TITLE
DLP-1 Registration

### PR DESCRIPTION
**Pull request by Balazs14-hun**

### CODE REVIEW

File: app/Http/Controllers/UserController.php [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  {
+    return Inertia::render('Users/Create');
+  }
+
+  /**
+   * Store a newly created resource in storage.
+   */
+  public function store(Request $request)
+  {
+    $attributes = $request->validate([
```
Comment (szabolcs3) [inline app/Http/Controllers/UserController.php:32]: Please use the validations [like this](https://laravel.com/docs/11.x/validation#creating-form-requests) 😄 . You don't need to define the rules here.

File: app/Http/Controllers/UserController.php [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+      'name' => ['required', 'max:255'],
+      'email' => ['required', 'email:strict', 'unique:users,email', 'max:255'],
+      'password' => ['required', 'min:8', 'max:255'],
+      'is_doctor' => ['required', 'boolean'],
+    ]);
+
+    User::create($attributes);
+
+    return redirect('/');
```
Comment (szabolcs3) [inline app/Http/Controllers/UserController.php:41]: Please use [named routes](https://laravel.com/docs/11.x/routing#named-routes)

File: resources/js/Components/BaseFormInput.vue [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        :id="id"
+        v-model="model"
+        class="mt-2 w-full rounded-lg border shadow-sm focus:border-primary-300 focus:ring-4 focus:ring-primary-200"
+        :class="[errorMessage ? 'border-error-300 caret-error-500' : 'border-gray-300']"
+        :type="inputType"
+        :name="id"
+        :placeholder="placeholder"
+        required
+      />
+      <ExclamationCircleIcon
```
Comment (szabolcs3) [inline resources/js/Components/BaseFormInput.vue:20]: Yes, you can do this or if you want to make it more beautiful. You can:
```
<slot name="icon" />
```

And you can overwrite it with
```
<template #icon>
</template>
```

[Read more](https://vuejs.org/guide/components/slots.html)

File: resources/js/Components/BaseSidePanel.vue [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+          />
+        </Link>
+        <slot />
+      </div>
+
+      <div class="m-10 flex flex-col justify-between text-sm text-primary-200 lg:flex-row">
+        <p class="">&copy; Delphi 2022</p>
+        <div class="flex items-center space-x-1.5">
+          <EnvelopeIcon class="size-4" />
+          <p class="">help@delphi.com</p>
```
Comment (szabolcs3) [inline resources/js/Components/BaseSidePanel.vue:19]: Empty class 😛

File: resources/js/Components/RegisterSidePanel.vue [hunk 5]
```diff
@@ -0,0 +1,24 @@
+<template>
+  <BaseSidePanel>
+    <RegisterSidePanelItem
+      title="Detaliile tale"
+      body="Intoduceți adresa de email și alegeți parola"
```
Comment (szabolcs3) [inline resources/js/Components/RegisterSidePanel.vue:5]: Missing `r` 😛

File: resources/js/Components/RegisterSidePanelItem.vue [hunk 6]
```diff
@@ -0,0 +1,47 @@
+<template>
+  <div
+    class="relative flex"
+    :class="[isFirst ? 'pb-4' : '', isLast ? 'pt-4' : '', !isLast && !isFirst ? 'pb-4 pt-4' : '']"
```
Comment (szabolcs3) [inline resources/js/Components/RegisterSidePanelItem.vue:4]: Check this [one](https://vuejs.org/guide/essentials/class-and-style#binding-html-classes) 😄

File: resources/js/Components/RegisterSidePanelItem.vue [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+  >
+    <CheckCircleIcon
```
Comment (szabolcs3) [inline resources/js/Components/RegisterSidePanelItem.vue:6]: I'm pondering whether this combination of v-if and v-else can doable using a mapper.

File: tailwind.config.js [hunk 8]
```diff
@@ -13,7 +16,119 @@ export default {
...
       fontFamily: {
-        sans: ['Figtree', ...defaultTheme.fontFamily.sans]
+        sans: ['Inter var', ...defaultTheme.fontFamily.sans]
+      },
+      spacing: {
+        13: '3.25rem'
+      },
+      colors: {
+        /* Primary colors */
+        primary: {
```
Comment (szabolcs3) [inline tailwind.config.js:26]: Nice 👏

File: tailwind.config.js [hunk 9]
```diff
@@ -13,7 +16,119 @@ export default {
   theme: {
     extend: {
       fontFamily: {
-        sans: ['Figtree', ...defaultTheme.fontFamily.sans]
+        sans: ['Inter var', ...defaultTheme.fontFamily.sans]
```
Comment (szabolcs3) [inline tailwind.config.js:19]: Just simply `Inter` no?

File: resources/js/Pages/Users/Create.vue [hunk 10]
```diff
@@ -0,0 +1,19 @@
+<template>
+  <Head title="Registration"
+    ><link
+      rel="stylesheet"
```
Comment (szabolcs3) [inline resources/js/Pages/Users/Create.vue:4]: I don't think this should be here 🤔

File: resources/js/Components/BaseContainer.vue
Comment (szabolcs3) [inline resources/js/Components/BaseContainer.vue:1]: Is this a layout for the website or just a container?

**Review by szabolcs3 — CHANGES_REQUESTED** [review]

File: resources/js/Components/BaseContainer.vue
Comment (Balazs14-hun) [reply resources/js/Components/BaseContainer.vue:1]: It's just a container for registration like cards. Login has the same container, so i thought i should extract it.

**Review by szabolcs3 — APPROVED** [review]

**Pull request merged**